### ISSUE TITLE
Fix Rubocop Rails issues

### ISSUE THREAD

**Issue by dacook** [openfoodfoundation/openfoodnetwork#11482]

**Comment by dacook** [comment]
Hi @macanudo527 , I see you've started on this so created an issue for tracking.

**Comment by macanudo527** [comment]
@dacook Can you assign this to me?

**Comment by dacook** [comment]
Hi @macanudo527 , how have you been? I hope you've had a fulfilling start to the year!
Just wondering if you'd like to pick this up this year, or if we should unassign for now?

**Comment by macanudo527** [comment]
@dacook Yeah, I've just been too busy with everything. I'd like to work on it, but just can't.

**Comment by dacook** [comment]
No problems, that's totally understandable. All the best for this season, and I hope to see you again soon!

**Comment by anthonyms** [comment]
@dacook Kindly assign this to me. 
Am I right in concluding that it is a subset of #6055?

**Comment by dacook** [comment]
Great, yep, that's correct Anthony. Good luck!

**Comment by sigmundpetersen** [comment]
@dacook maybe we could create sub-issues of the task list in the description?
Then each issue will check itself when a PR is merged towards it.

We can create them as we go. Just click to the far right of each checkbox.

**Comment by dacook** [comment]
Thanks Sigmund. Yes I hadn't bothered with this because I wasn't sure how big each of the items was. It's possible some are so small that they don't need to have a separate issue/pr..

**Comment by Nwabor** [comment]
@dacook I can work on some of this. Can it be assigned to me?

**Comment by Nwabor** [comment]
@sigmundpetersen I can work on this, can it be assigned to me?

**Comment by sigmundpetersen** [comment]
Hi and welcome @Nwabor 👋 
Yes, thank you. You can work on this.
Please pick one of the open bullets in the Issue list in the description that doesn't have a PR attached.

**Comment by Nwabor** [comment]
Thank you @sigmundpetersen.

**Comment by Nwabor** [comment]
@sigmundpetersen i need access right to push my PR.

**Comment by rioug** [comment]
@Nwabor you should not need right to create the PR, have a look at the documentation here : https://github.com/openfoodfoundation/openfoodnetwork/wiki/Making-a-great-pull-request . Make sure you are using your own fork.

**Comment by Nwabor** [comment]
@sigmundpetersen can I pick another task, while waiting for the PR I pushed to be reviewed?

**Comment by sigmundpetersen** [comment]
Yes, go ahead @Nwabor 👍

**Comment by Nwabor** [comment]
Okay thanks, I will resolve the conflicts as well @sigmundpetersen

**Comment by cyrillefr** [comment]
Hello @dacook ,

The [Rails/SquishedSQLHeredocs](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railssquishedsqlheredocs)  item is not in the `.rubocop_todo.yml` file anymore.
It has been fixed in 2023. It can be removed from the list.

Regarding the [Rails/TransactionExitStatement](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railstransactionexitstatement), the offense will no longer be effective in Rails 7.2 upwards since the behaviour that has changed will be restored. (I don't know how to tag this item).

**Comment by dacook** [comment]
Thanks Cyrille, that's interesting, the list was last generated in 2024, I don't know why SquishedSQL was included then. 🤷 
 I've ticked SquishedSQL and crossed out TransactionExitStatement with a note about 7.2.

Thanks to your latest contribution, we're almost finished!

**Comment by dacook** [comment]
Hi @cyrillefr , just wondering if you are planning to do any further work on this? If so I will assign the issue to you to avoid double-up with anyone else.

**Comment by cyrillefr** [comment]
Hello @dacook ,

The [Rails/LexicallyScopedActionFilter](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Rails/LexicallyScopedActionFilter) box can be checked with a reference to #13251 instead :)

That leaves this issue with the #13163 which is blocked and it will be done.

**Comment by cyrillefr** [comment]
Hello @sigmundpetersen , 

the [Rails/UniqueValidationWithoutIndex](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Rails/UniqueValidationWithoutIndex) box may be ticked now.

**Comment by sigmundpetersen** [comment]
Ok @dacook I guess we can close this one then? 
The remaining issue will be gone after Rails 7.2.

**Comment by mkllnk** [comment]
I can confirm that all safe cops have been dealt with:

```
grep " safe autocorrection" .rubocop_todo.yml
# no output
```

We still got a few other rules though:

```
$ grep '^[^ #]' .rubocop_todo.yml 
Metrics/AbcSize:
Metrics/BlockLength:
Metrics/BlockNesting:
Metrics/ClassLength:
Metrics/CyclomaticComplexity:
Metrics/MethodLength:
Metrics/ModuleLength:
Metrics/ParameterLists:
Metrics/PerceivedComplexity:
Rails/TransactionExitStatement:
Rails/UniqueValidationWithoutIndex:
Style/ClassAndModuleChildren:
Style/MapToHash:
Style/MissingRespondToMissing:
Style/OpenStructUse:
Style/OptionalBooleanParameter:
```

I wouldn't bother with the Metrics cops. They should be dealt with over time when touching the code anyway. One Rails cop will become obsolete and the other one was actually dealt with. The entries were just not removed from the todo list. Some of the Style rules could be good to deal with but I'm worried that we may accidentally break something along the way because they don't have safe auto-correction.

Anyway, this issue can be closed and we can have a look at other cops in the parent issue:

* #6055
